docs/CODE_GUIDELINES.md

```diff
@@ -480,7 +480,7 @@ EDS adheres to the following API naming conventions:
 
 ### Text, Labels, Titles
 
-- Default to `text` for short strings of text, such as `<Button text="Click here">` or `<Badge text="Overdue" />`.
+- Default to `text` for short strings of text, such as `<BreadCrumbsItem text="My Courses">` or `<Badge text="Overdue" />`.
```

Comment:
I just picked another component that still uses `text`. I dunno if we want to add something in the docs about the `children` paradigm, which is now being used by `Button` and `Text`.

Reply:
@dierat if I understand correctly, your concern stems from the way in which the Button component handles its contents; i.e. it is being switched from this pattern:

`<Button>Press me</Button>`

to this pattern:

`<Button text="Press me" />`

and not just the prop name itself being `text`. Basically now you're taking a component that held content between its opening and closing tags and making devs relearn it as a self-closing tag that only accepts named props.

JSX (and by extension, TypeScript) reserve the word "children" to be a catch-all term for... the kitchen sink. Every React component has access to `props.children` containing the contents of anything between the opening and closing tags. Unless you specify otherwise, TypeScript makes assumptions that children can be anything, including another React component. Is that the intention? Or do the contents of the button component need to be constrained to a text string only? I'm assuming the latter, but I shouldn't assume.

Reply:
I should have added: one of the changes that I put in the TypeScript section of CODE_GUIDELINES was to move components away from using `React.FC` as their source of typing, and the reasoning behind that change was actually to reign in the free-for-all use of `children`. Components that are typed with `React.FC` receive implicit permission to receive `children`... they are typed for you as `ReactNode` even if you don't explicitly include them in your explicit typing of Props. Taking that off of the components forces us to now explicitly type `children`, which is what you would need to do as part of this ticket, and going forward, devs would have to be reminded that `children` on buttons is treated differently and needs to be typed as a string.

Reply:
I think it'd be good to include something about `children` and when we'd prefer that

@csnizik great points 🙂 Our main motivation is that we _want_ to allow more than just strings in buttons. We've seen cases in the Learning Platform where the extra flexibility is needed, and we're concerned that making the API tighter (to only allow strings) would hinder adoption. So we're okay with the more relaxed typing of `ReactNode` here

Reply:
@anniehu4 so for instance if someone wants a button with an icon instead of text, or an icon positioned before/after a text string, you want them to have the flexibility to do that? I'm not sure that the flexibility of allowing children of type `ReactNode` would give them the results you want style-wise. i.e., how do you set up button styles to properly display any component or html tag that gets passed in?

Reply:
yep, exactly! That's how our current edu-design-system button does it. We set up some padding / icon sizing styles to ensure things look good by default ([see code file here](https://github.com/chanzuckerberg/edu-design-system/blob/main/packages/components/src/ClickableStyle/ClickableStyle.module.css#L16)) -- @dierat probably knows better the exact relevant styles, if you're curious.

We have a good amount of usage of the Buttons in the platform now and haven't run into styling complaints yet, so we feel pretty good about this approach. Will note that all the usage has been passing a string or an icon with a string -- so we don't anticipate people doing things that crazy with it, but do want to allow for the existing usages

Reply:
Something I just realized is that the `text` prop is already accepting a `ReactNode` in the type, so the argument that we should keep `text` in order to restrict the type doesn't make much sense to me. If the type is the same, shouldn't we just pick the name and structure that makes sense for our team?

https://github.com/chanzuckerberg/edu-design-system/blob/next/src/components/Button/Button.tsx#L60